src/python/pants/engine/target.py

```diff
@@ -838,18 +838,20 @@ class CoarsenedTargetsRequest:
 
     roots: Tuple[Address, ...]
     expanded_targets: bool
-    include_special_cased_deps: bool
+    should_traverse_deps_predicate: ShouldTraverseDepsPredicate
 
     def __init__(
         self,
         roots: Iterable[Address],
         *,
         expanded_targets: bool = False,
-        include_special_cased_deps: bool = False,
+        should_traverse_deps_predicate: ShouldTraverseDepsPredicate | None = None,
     ) -> None:
         object.__setattr__(self, "roots", tuple(roots))
         object.__setattr__(self, "expanded_targets", expanded_targets)
-        object.__setattr__(self, "include_special_cased_deps", include_special_cased_deps)
+        if should_traverse_deps_predicate is None:
+            should_traverse_deps_predicate = should_traverse_deps_default_predicate
```

Comment:
(You could inline this in the `__setattr__` call, same below)
```suggestion
        should_traverse_deps_predicate = should_traverse_deps_predicate or should_traverse_deps_default_predicate
```